### PR TITLE
Change home-robot to optional dep

### PR DESCRIPTION
**Pull request by exhaustin**
## Motivation and Context

Change [home_robot](https://github.com/facebookresearch/home-robot/tree/main/src/home_robot) to be an optional dep.

TODO: Modify documentation and challenge description to inform users that continuous actions require installing with the "continuous control flag, as in
```
pip install -e "habitat-lab[continuous_control]"
```

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ykarmesh** [comment]
Can you also update the README to tell users about this and how to install this additional dependency?

File: habitat-lab/habitat/tasks/nav/nav.py [hunk 1]
```diff
@@ -1537,7 +1536,16 @@ def __init__(self, *args: Any, **kwargs: Any):
...
+        try:
+            from habitat.utils.controller_wrapper import ContinuousController
+
+            self.w2v_controller = ContinuousController(self._config)
+        except ModuleNotFoundError:
+            print(
+                "Missing dependencies for waypoint type actions. "
+                "Install habitat-lab with the 'continuous_control' option to enable this feature."
+            )
+            raise
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/nav/nav.py:1548]:
```suggestion
        except ModuleNotFoundError as exc:
            additional_error_message = """
            Missing dependencies for waypoint type actions. 
            Install habitat-lab with the 'continuous_control' option to enable this feature.
            pip install -e "habitat-lab[continuous_control]
            """
            exc.message += additional_error_message
            raise exc
```

Comment (vincentpierre) [reply habitat-lab/habitat/tasks/nav/nav.py:1548]: Please make sure this error is raised properly and that the message looks good. No need for a unit test but we need to make sure this works.

File: habitat-lab/habitat/tasks/nav/nav.py [hunk 2]
```diff
@@ -1537,7 +1536,16 @@ def __init__(self, *args: Any, **kwargs: Any):
         super().__init__(*args, **kwargs)
 
         # Init goto velocity controller
-        self.w2v_controller = ContinuousController(self._config)
+        try:
+            from habitat.utils.controller_wrapper import ContinuousController
+
+            self.w2v_controller = ContinuousController(self._config)
```
Comment (vincentpierre) [inline habitat-lab/habitat/tasks/nav/nav.py:1542]: ContinuousController is a very vague name at this API level. Please change the name to a more descriptive one.

File: habitat-lab/setup.py [hunk 3]
```diff
@@ -36,6 +36,13 @@
     "Bug Tracker": "https://github.com/facebookresearch/habitat-lab/issues",
 }
 
+EXTRA_DEPS = {
+    # Home robot dependency for velocity controllers
+    "continuous_control": [
```
Comment (vincentpierre) [inline habitat-lab/setup.py:41]: Why call it continuous_control ? 
All rearrangement tasks are using continuous control and none of them use home-robot. 
Suggestions : 
 - continuous_navigation
 - home-robot
 - continuous_control_for_navigation

Comment (ykarmesh) [reply habitat-lab/setup.py:41]: We might add more controllers from home-robot in the future so using home_robot is accurate in my opinion

Comment (ykarmesh) [inline habitat-lab/setup.py:41]:
```suggestion
    "home_robot": [
```

File: README.md [hunk 4]
```diff
@@ -79,6 +79,12 @@ If you use the Habitat platform in your research, please cite the [Habitat 1.0](
       cd habitat-lab
       pip install -e habitat-lab  # install habitat_lab
       ```
+
+      Note: To enable continuous robot base action spaces, install habitat-lab with the "continuous_control" option
+      ```
+      pip install -e "habitat-lab[continuous_control]"
+      ```
+
```
Comment (ykarmesh) [inline README.md:87]:
```suggestion
      Note: To enable continuous robot base action spaces, install habitat-lab with the "home_robot" option
      ```
      pip install -e "habitat-lab[home_robot]"
      ```

```

File: habitat-lab/habitat/tasks/nav/nav.py (hunk 2 above)
Comment (exhaustin) [reply habitat-lab/habitat/tasks/nav/nav.py]: Changed it to `DiffDriveVelocityController`. Let me know if there's a better one!

File: habitat-lab/habitat/tasks/nav/nav.py [hunk 5]
```diff
@@ -1537,7 +1536,19 @@ def __init__(self, *args: Any, **kwargs: Any):
...
+            from habitat.utils.controller_wrapper import (
+                DiffDriveVelocityController,
+            )
+
+            self.w2v_controller = DiffDriveVelocityController(self._config)
+        except ModuleNotFoundError as exc:
+            additional_error_message = """
+            Missing dependencies for waypoint type actions.
+            Install habitat-lab with the 'continuous_control' option to enable this feature.
+            pip install -e "habitat-lab[continuous_control]
```
Comment (ykarmesh) [inline habitat-lab/habitat/tasks/nav/nav.py:1549]:
```suggestion
            Install habitat-lab with the 'home_robot' option to enable this feature.
            pip install -e "habitat-lab[home_robot]
```

**Review by ykarmesh — APPROVED** [review]

**Pull request merged**